src/openrct2/ReplayManager.cpp

```diff
@@ -0,0 +1,773 @@
+                {
+                    actionType = command.action->GetType();
+                }
+                serialiser << actionType;
+
+                if (serialiser.IsLoading())
+                {
+                    command.action = GameActions::Create(actionType);
+                    Guard::Assert(command.action != nullptr);
+                }
+
+                command.action->Serialise(serialiser);
+            }
+            else
+            {
+                serialiser << command.eax;
+                serialiser << command.ebx;
+                serialiser << command.ecx;
+                serialiser << command.edx;
+                serialiser << command.esi;
+                serialiser << command.edi;
+                serialiser << command.ebp;
+                serialiser << command.callback;
+            }
+            return true;
+        }
+
+        bool Serialise(DataSerialiser& serialiser, ReplayRecordData& data)
+        {
+            serialiser << data.name;
+            serialiser << data.parkData;
+            serialiser << data.parkParams;
+            serialiser << data.spriteSpatialData;
+            serialiser << data.tickStart;
+            serialiser << data.tickEnd;
+
+            uint32_t countCommands = (uint32_t)data.commands.size();
+            serialiser << countCommands;
+
+            if (serialiser.IsSaving())
+            {
+                for (auto& command : data.commands)
+                {
+                    SerialiseCommand(serialiser, const_cast<ReplayCommand&>(command));
+                }
+            }
+            else
+            {
+                for (uint32_t i = 0; i < countCommands; i++)
+                {
+                    ReplayCommand command = {};
+                    SerialiseCommand(serialiser, command);
+
+                    data.commands.emplace(std::move(command));
+                }
+            }
+
+            uint32_t countChecksums = (uint32_t)data.checksums.size();
+            serialiser << countChecksums;
+
+            if (serialiser.IsLoading())
+            {
+                data.checksums.resize(countChecksums);
+            }
+
+            for (uint32_t i = 0; i < countChecksums; i++)
+            {
+                serialiser << data.checksums[i].first;
+                serialiser << data.checksums[i].second.raw;
+            }
+
+            return true;
+        }
+
+#ifndef DISABLE_NETWORK
+        void CheckState()
+        {
+            uint32_t checksumIndex = _currentReplay->checksumIndex;
+
+            if (checksumIndex >= _currentReplay->checksums.size())
+                return;
+
+            const auto& savedChecksum = _currentReplay->checksums[checksumIndex];
+            if (_currentReplay->checksums[checksumIndex].first == gCurrentTicks)
+            {
+                rct_sprite_checksum checksum = sprite_checksum();
+                if (savedChecksum.second.ToString() != checksum.ToString())
```

Comment:
It should not be necessary to compare the stringified checksums, rather compare them directly.